src/vos/tests/SConscript

```diff
@@ -29,8 +30,41 @@ def scons():
     vos_size = daos_build.program(denv, 'vos_size', ['vos_size.c'],
                                   LIBS=libraries)
 
-    denv.Install('$PREFIX/bin/', [vos_tests, vos_size, evt_ctl, 'vos_size.py'])
-    denv.Install('$PREFIX/etc/', ['vos_size_input.yaml', 'vos_dfs_sample.yaml'])
+    # Build the vos_size_dfs_sample binary. It will run on the build system
+    dlenv = denv.Clone()
+    dlenv.AppendUnique(LIBPATH=['$BUILD_DIR/src/cart/src/cart'])
+    dlenv.AppendUnique(LIBPATH=['$BUILD_DIR/src/cart/src/gurt'])
+    dlenv.AppendUnique(LIBPATH=['$BUILD_DIR/src/client/api'])
+    dlenv.AppendUnique(LIBPATH=['$BUILD_DIR/src/client/dfs'])
+    dlenv.AppendUnique(LIBPATH=['$BUILD_DIR/src/common'])
+
+    libraries = ['daos', 'daos_common', 'dfs', 'gurt', 'cart', 'pmemobj']
+
+    dfs_sample = daos_build.program(dlenv, 'vos_size_dfs_sample',
+                                    ['vos_size_dfs_sample.c'], LIBS=libraries)
+
+    # Create the VOS sample yaml file with the DFS layout
+    dlenv['DFS_SAMPLE_DIR'] = '$BUILD_DIR/src/vos/tests'
+
+    run_dfs_sample = ['LD_LIBRARY_PATH=$LD_LIBRARY_PATH:'
```

Comment:
LD_LIBRARY_PATH should not be needed here.

Reply:
Without the LD_LIBRARY_PATH set, the rpmbuild command complains about missing libraries.

`error while loading shared libraries: libdaos.so.0: cannot open shared object file: No such file or directory`
Full log [here](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-2947/5/artifact/artifacts/centos7/build.log)

I tried different approaches such as using:
`dlenv.AppendUnique(LIBPATH=['$BUILD_DIR/src/client/api'])`
`dlenv.AppendUnique(RPATH_FULL=['../../client/api'])`
`dlenv.AppendUnique(RPATH_FULL=['$BUILD_DIR/src/client/api'])`

I also tried with the installed binary, however, by the time rpmbuild try to run it. It doesn't exist yet.
`$PREFIX/bin/vos_size_dfs_sample -f $DFS_SAMPLE_DIR/vos_dfs_sample.yaml`

All of them failed. I appreciate any suggestion regarding how to handle this. LD_LIBRARY_PATH is not persistent; it is only used to generate the vos_size_input.yaml file on the build system.

Reply:
@jolivier23 I think that it might not be possible to remove the LD_LIBRARY_PATH. I have this other [draft](https://github.com/daos-stack/daos/pull/2988/files) without the LD_LIBRARY_PATH. The command:
`scons --config=force --build-deps=yes TARGET_TYPE=dev install`
works good when run locally. However, when the CI runs the `rpmbuild` command, it fails. [Full log](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-2988/8/artifact/artifacts/centos7/build.log)

Reply:
You should not need it.  I think a lot of your problems will go away and the build would be significantly simpler if you placed this file in client/dfs/tests instead...that way you can inherit the dfs environement which will have all the necessary include paths.    Or maybe even in client/dfs since it's not really a test but rather a utility we run to generate the data for vos size.   It's not really related to VOS at all